Sources/Amplitude/Plugins/iOS/UIKitUserInteractions.swift

```diff
@@ -0,0 +1,189 @@
+    }
+
+    func extractData(with action: Selector) -> ViewData {
+        var targetText: String?
+
+        if let button = self as? UIButton {
+            targetText = button.currentTitle
+        } else if let segmentedControl = self as? UISegmentedControl {
+            targetText = segmentedControl.titleForSegment(at: segmentedControl.selectedSegmentIndex)
+        } else {
+            #if !os(tvOS)
+            if #available(iOS 14.0, macCatalyst 14.0, *) {
+                if let colorWell = self as? UIColorWell {
+                    targetText = colorWell.title
+                } else if let `switch` = self as? UISwitch {
+                    targetText = `switch`.title
+                }
+            }
+            #endif
+        }
+
+        let viewController = owningViewController
+        let viewControllerClassName = viewController?.descriptiveTypeName
+        let viewControllerTitle = viewController?.title
+        let targetAccessibilityLabel = self.accessibilityLabel
+        let actionName = NSStringFromSelector(action)
+        let targetViewClassName = self.descriptiveTypeName
+        let viewHierarchy = sequence(first: self, next: { $0.superview })
+            .map { $0.descriptiveTypeName }
+            .joined(separator: UIView.viewHierarchyDelimiter)
+
+        return ViewData(
+            viewController: viewControllerClassName,
+            title: viewControllerTitle,
+            accessibilityLabel: targetAccessibilityLabel,
+            actionMethod: actionName,
+            targetViewClass: targetViewClassName,
+            targetText: targetText,
+            hierarchy: viewHierarchy)
+    }
+}
+
+extension UIResponder {
+    var descriptiveTypeName: String {
+        String(describing: type(of: self))
+    }
+
+    var owningViewController: UIViewController? {
+        return self as? UIViewController ?? next?.owningViewController
+    }
+}
+
+protocol Trackable {
+    func shouldTrack(_ action: Selector, for event: UIEvent?) -> Bool
+}
+
+extension UITextField: Trackable {
+    func shouldTrack(_ action: Selector, for event: UIEvent?) -> Bool {
+        false
+    }
+
+    @objc static func amp_textFieldBeganEditing(_ notification: NSNotification) {
+        guard let textField = notification.object as? UITextField else { return }
+        let userInteractionEvent = textField.eventFromData(with: #selector(UITextField.amp_textFieldBeganEditing))
+        UIKitUserInteractions.amplitudeInstances.allObjects.forEach {
+            $0.track(event: userInteractionEvent)
+        }
+    }
+
+    @objc static func amp_textFieldEndedEditing(_ notification: NSNotification) {
+        guard let textField = notification.object as? UITextField else { return }
+        let userInteractionEvent = textField.eventFromData(with: #selector(UITextField.amp_textFieldEndedEditing))
+        UIKitUserInteractions.amplitudeInstances.allObjects.forEach {
+            $0.track(event: userInteractionEvent)
+        }
+    }
```

Comment:
Anything objc/dynamic is basically in a global namespace on objects, so we should avoid adding API surface area here whenever possible. Since these are just notification listeners, it should be pretty simple to add them to one of our owned classes.